plugins/search/src/components/SearchType/SearchType.tsx

```diff
@@ -56,6 +58,9 @@ export type SearchTypeProps = {
   defaultValue?: string[] | string | null;
 };
 
+/**
+ * @public
+ */
 const SearchType = (props: SearchTypeProps) => {
   const { className, defaultValue, name, values = [] } = props;
```

Comment:
It took me a second to remember why I'd omitted `SearchType` from the list of components to migrate over to `@backstage/plugin-search-react`...  And I just did the same bit of research / thinking again:

As it is today, the only place _likely_ to use the `<SearchType>` component (or its variants like the accordion or the tabs), is in the app when composing _the_ search page or _the_ modal.  Because it's the app, it's fine for it to be exported from the plugin.  In which case, perhaps adding a comment here (within the function body, not associated with the `@public`) would be good as a reminder.

...I suppose it's still _possible_ that a plugin would want to operate on the `types` portion of the search context (e.g. if a single plugin was responsible for multiple indexes/collators and wanted a unified search experience over those types _within the plugin experience_).  ...And in that case, perhaps we'd want to move it out to `@backstage/plugin-search-react` as well?  And in that case, we may as well do it now while we're updating `create-app` and others.

I dunno.  Would be curious to hear your thoughts on this, @emmaindal.

Reply:
I think we do see a need for plugin focused search experiences and we would like to provide a way to compose your own search page for your own plugin. With that said, I don't see a reason to why you would like to use the `SearchType` component on that plugin search page, as you probably shouldn't let the user search across other types in that use case (and thats why you also probably would like to keep your indicies on a much higher level and then use fields with metadata that you can narrow down the search on) So instead you would use something like the [SearchFilter](https://backstage.io/storybook/?path=/story/plugins-search-searchfilter--check-box-filter) for other kind of filters.

Reply:
I think until we really see a need to use the `SearchType` component anywhere else than in the app and a good reason for that, we can keep it in plugin-search. What do you think?